### DISCUSSION TITLE
Results: Forces on edges of 2D members, what should be defined?

### DISCUSSION THREAD

**Discussion by scia-nv** [StructuralAnalysisFormat/StructuralAnalysisFormat-Doc#25] [Ideas]
Results: Forces on edges of 2D members, what should be defined?

**Comment by robinblazek** [comment]
Hi all, 

after our analysis we would like to present you the first proposal of the results on the edge of 2D member. We tried to make it simple as possible and also keep our existing principles. You can see, that the most of it it's the same as for results on 1D member. We see it as the right way, because in principle result on the edge is very similar to result on 1D member. 

It is just first draft, so don't hesitate and let us know your opinions. We are open to your ideas for changes and improvements. If you have any questions, ask us anytime. 
Thank You!

**There is link to draft excel with results on edges of 2D member:**
[SAF-example](https://scia.sharepoint.com/:x:/s/Calatrava/EWsfbznwaH5NjAstI5XJ9KABAubaa6X7hTiLCUIodP4zFA?e=rP4SNe)

**Object scheme in analysis SW:**
![image](https://user-images.githubusercontent.com/88313526/141445464-b2681cdf-7906-4b01-9edb-7f85f3a73799.png)

**Header of columns for the new object in SAF:**
![image](https://user-images.githubusercontent.com/88313526/141445784-205979d3-8533-411c-b597-4085f837f9fe.png)

**Explanation of the properties:**

| Property         | Description                                                                                                             |
|------------------|-------------------------------------------------------------------------------------------------------------------------|
| Result on        | Specify object, where the result is on                                                                                  |
| 2D member        | Name of the 2D member, whose edge is the result on                                                                      |
| Edge             | Number of the edge                                                                                                      |
| Result for       | Specifies from where the result is coming from (from Load Case, Load Combination)                                       |
| Load case        | Reference to the name of the load case                                                                                  |
| Load combination | Reference to the name of the load combination                                                                           |
| Combination key  | Allows to define exact combination per result section                                                                   |
| Section at [m]   | X coordinate (LCS of the edge) on the edge (distance from the start node) where the result is located                   |
| Index            | Defines an order of the section on the edge, starting with 1 and increasing from the start to the end node of the edge. |
| mx [kNm/m]       | Bending moment mx                                                                                                       |
| my [kNm/m]       | Bending moment my                                                                                                       |
| mxy [kNm/m]      | Torsion moment mxy                                                                                                      |
| vx [kN/m]        | Shear force vx                                                                                                          |
| vy [kN/m]        | Shear force vy                                                                                                          |
| nx [kN/m]        | Membrane force nx                                                                                                       |
| ny [kN/m]        | Membrane force ny                                                                                                       |
| nxy [kN/m]       | Shear force nxy                                                                                                         |

**Reply by andreasniggl** [reply]
Hi, I'm wondering about the coordinate system the forces refer to. Are the forces to be interpreted in the coordinate system of the 2D member or according to a coordinate system along the edge. For example, if one wants to extract (and probably sum up) the membrane forces acting perpendicular to a - possibly inclined -  edge.  Is it needed to transform the [nx,ny,nxy] tensor or can just the 'nx' or 'ny' component be taken?

**Reply by robinblazek** [reply]
Hi, thanks for a very good question. Our proposal is to have the LCS of the results the same as the LCS of the 2D member. So yes, for perpendicular forces to inclined edges transformation would be needed. 

There were two reasons/ideas, why we chose this:
1) We think, that if it is the same directions as the LCS of 2D member, then it is more general/universal solution. 2D results on the edges can be used for various types of applications and at some point there always be needed transformation for some of the cases. This type of result seems to us very clear and possible to use for further various transformations.
2) We also think, that if we in the future will extend results for the whole 2D member mesh (not only for the edges), that it should correspond to the way we represent results on the edges. Because it is actually still the same results - results for 2D member. Results on the edges are just "special" subgroup of it (at least from our point of view). 

That's pretty much it, why we thought, that it is the right way. 
What's your opinion about that? Is it OK for you or would you prefer to be done differently? If you have any ideas or problematic cases where our solution is not correct, please, definitely share it with us. We are really open to discuss it. As I mentioned, we also try to think about possible future extensions and for that it is always really important to do the "first steps" correctly, because then it is much harder to make change. 

Thank you very much for getting involved!

**Comment by sonkaste17** [comment]
Introduced in dev (latest) version of SAF specification:
https://www.saf.guide/en/latest/results/resultinternalforce2dedge.html
